lib/data/school_importer.rb

```diff
@@ -0,0 +1,78 @@
+require 'breasal'
+
+class SchoolImporter
+  attr_accessor :urns, :edubase_data
+  def initialize(urns, edubase_data)
+    self.urns = urns
+      .reject { |l| l.in?(%w{URN SCITT TRUST TSA}) }
+      .map(&:strip)
+      .map(&:to_i)
+
+    self.edubase_data = edubase_data
+      .each
+      .with_object({}) do |record, h|
+        h[record['URN'].to_i] = record
+      end
+  end
+
+  def import
+    total = @urns.length
+
+    Bookings::School.transaction do
+      @urns.each.with_index(1) do |urn, i|
+        unless (row = @edubase_data[urn])
+          raise "URN #{urn} cannot be found in dataset"
+        end
+
+        if build_school(row).save
+          if Rails.env != 'test'
+            puts("%<count>s of %<total>d | %<urn>s | %<name>s" % {
+              count: i.to_s.rjust(3),
+              total: total,
+              urn: urn.to_s.rjust(8),
+              name: row['EstablishmentName']
+            })
+          end
+        else
+          fail "failed to import #{urn}"
+        end
+      end
+    end
+  end
+
+private
+
+  def build_school(row)
+    Bookings::School.new(
+      urn:          nilify(row['URN']),
+      name:         nilify(row['EstablishmentName']),
+      website:      nilify(row['SchoolWebsite']),
+      address_1:    nilify(row['Street']),
+      address_2:    nilify(row['Locality']),
+      address_3:    nilify(row['Address3']),
+      town:         nilify(row['Town']),
+      county:       nilify(row['County (name)']),
+      postcode:     nilify(row['Postcode']),
+      coordinates:  convert_to_point(row['Easting'], row['Northing'])
+    ).tap do |school|
+      school.phases << phases[row['PhaseOfEducation (code)'].to_i]
+    end
+  end
+
+  def nilify(val)
+    val.present? ? val : nil
+  end
+
+  def phases
+    @phases ||= Bookings::Phase.all.group_by(&:edubase_id)
```

Comment:
Shouldn't this be `index_by` ?

This will result in 

```ruby
phases = {
  1 => [Bookings::Phase]
  2 => [Bookings::Phase]
}
```

instead of 

```ruby
phases = {
  1 => Bookings::Phase
  2 => Bookings::Phase
}
```

Reply:
Yes, both work although in this case `.index_by` is definitely cleaner.